front/pages/w/[wId]/ds/index.tsx

```diff
@@ -54,8 +148,67 @@ export default function DataSourcesView({
   owner,
   readOnly,
   dataSources,
+  managedDataSources,
+  canUseManagedDataSources,
   gaTrackingId,
+  nangoPublicKey,
+  nangoSlackConnectorId,
+  nangoNotionConnectorId,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const [managedDataSourcesLocal, setManagedDataSourcesLocal] =
+    useState(managedDataSources);
+
+  const [isLoadingByProvider, setIsLoadingByProvider] = useState<
+    Record<ConnectorProvider, boolean | undefined>
+  >({} as Record<ConnectorProvider, boolean | undefined>);
+
+  const handleEnableManagedDataSource = async (provider: ConnectorProvider) => {
+    const nangoConnectorId =
+      provider == "slack" ? nangoSlackConnectorId : nangoNotionConnectorId;
+    const nango = new Nango({ publicKey: nangoPublicKey! });
+    const {
+      connectionId: nangoConnectionId,
+    }: { providerConfigKey: string; connectionId: string } = await nango.auth(
+      nangoConnectorId!,
+      `${provider}-${owner.sId}`
+    );
+
+    setIsLoadingByProvider((prev) => ({ ...prev, [provider]: true }));
+    try {
+      const res = await fetch(`/api/w/${owner.sId}/data_sources/managed`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          provider,
+          nangoConnectionId,
+        }),
+      });
+      if (res.ok) {
+        setManagedDataSourcesLocal((prev) =>
+          prev.map((ds) =>
+            ds.connectorProvider == provider
+              ? { ...ds, enabled: true, status: "succeeded" }
+              : ds
+          )
+        );
+      } else {
```

Comment:
@spolu  any idea what we should be doing here ? Not sure how I can really surface the issue to the user -- for now I was thinking of just assuming it's going to work

Reply:
also a window.alert will go a long way for the user